// This file was generated with the Protobuf generator tool.
// Dataset ID    : 40664
// Dataset Name  : car-evaluation;
// Dataset URL   : https://www.openml.org/data/v1/download/4965239/car-evaluation.arff
// Num. Columns  : 22
// Num. Rows     : 1728
// Target Feature: class

// Beginning of Description of Dataset:
// car-evaluation-pmlb
//

syntax = "proto3";

message Empty {
}

message Features {
	string   Class                          = 1;
	string   Buying_price_vhigh             = 2;
	string   Buying_price_high              = 3;
	string   Buying_price_med               = 4;
	string   Buying_price_low               = 5;
	string   Maintenance_price_vhigh        = 6;
	string   Maintenance_price_high         = 7;
	string   Maintenance_price_med          = 8;
	string   Maintenance_price_low          = 9;
	string   Doors_2                        = 10;
	string   Doors_3                        = 11;
	string   Doors_4                        = 12;
	string   Doors_5more                    = 13;
	string   Persons_2                      = 14;
	string   Persons_4                      = 15;
	string   Persons_more                   = 16;
	string   Luggage_boot_size_small        = 17;
	string   Luggage_boot_size_med          = 18;
	string   Luggage_boot_size_big          = 19;
	string   Safety_low                     = 20;
	string   Safety_med                     = 21;
	string   Safety_high                    = 22;
}

service get_next_row {
	 rpc get_next_row(Empty) returns(Features);
}

/*
//This code section can be directly used
//at the end of gen_next_row in the server.py file
//for OpenML dataset nr. 40664
        response.Class                          = row[0]
        response.Buying_price_vhigh             = row[1]
        response.Buying_price_high              = row[2]
        response.Buying_price_med               = row[3]
        response.Buying_price_low               = row[4]
        response.Maintenance_price_vhigh        = row[5]
        response.Maintenance_price_high         = row[6]
        response.Maintenance_price_med          = row[7]
        response.Maintenance_price_low          = row[8]
        response.Doors_2                        = row[9]
        response.Doors_3                        = row[10]
        response.Doors_4                        = row[11]
        response.Doors_5more                    = row[12]
        response.Persons_2                      = row[13]
        response.Persons_4                      = row[14]
        response.Persons_more                   = row[15]
        response.Luggage_boot_size_small        = row[16]
        response.Luggage_boot_size_med          = row[17]
        response.Luggage_boot_size_big          = row[18]
        response.Safety_low                     = row[19]
        response.Safety_med                     = row[20]
        response.Safety_high                    = row[21]
*/

